src/data/openthaigpt_pretraining_data/dedup_minhashlsh.py

```diff
@@ -0,0 +1,120 @@
+from datasketch import MinHash, MinHashLSH
+import numpy as np
+
+
+# create minhash signature
+def generate_minhash_signature(text, num_perm=128):
+    """
+    convert text string to minhash
+    """
+    minhash = MinHash(num_perm=num_perm)
+    shingles = [text[i : i + 3] for i in range(len(text) - 2)]
+    shingles = set(shingles)
+    for shingle in shingles:
+        minhash.update(shingle.encode("utf-8"))
+    return minhash
+
+
+# create minhasLSH from text_lists
+def generate_minhashlsh(text_list, threshold=0.9, num_perm=128):
+    """
+    create MinhashLSH from [text_list] --> lsh_dict
+    - threshold is jaccard number. Default is 0.9 which can
+    detect NearDup which is slightly changed in a few word.
+    - For detecting ExactDup, recommend threshold=0.97
+    - num_perm is number of permutation. Default is 128.
+    Higher number, better accuracy,
+    but trade-off with slower speed and higher memory
+    """
+    minhash_list = [generate_minhash_signature(t) for t in text_list]
+    lsh = MinHashLSH(threshold=threshold, num_perm=num_perm)
+    for i in range(len(minhash_list)):
+        lsh.insert(i, minhash_list[i])
+
+    lsh_dict = {"lsh": lsh, "text_list": text_list, "minhash_list": minhash_list}
```

Comment:
Please use CONST in `lsh`, `text_list`, and `minhash_list`.

Reply:
done ka